lib/saml_idp/service_provider.rb

```diff
@@ -19,6 +19,10 @@ def valid?
       attributes.present?
     end
 
+    def valid_identifier?
+      valid? && !identifier.nil?
```

Comment:
According to the `spec/lib/saml_idp/service_provider_spec.rb` tests, a ServiceProvider is valid if it has (a) a fingerprint and (b) a metadata_url. The `identifier` is only relevant in the context of the Request, not when a ServiceProvider is used in isolation.